web/src/lib/components/NostrButton.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+    import {onDestroy} from "svelte";
+    import {Pool} from "$lib/nostr/pool";
+    import {getChannelIdForStallOwner, pmChannelNostrRoomId} from '$lib/nostr/utils'
+    import {user} from "$lib/stores";
+
+    export let pmURL: string | null;
+
+    const pool: Pool = new Pool();
+
+    function postToNostr(location) {
+        let message;
+
+        let nostrRoomId;
+        switch (location) {
+            case 'stall':
+                message = 'Hi people! I just created a new product. Give it a look:';
```

Comment:
You're right. I'll change it, but those messages need to be rewritten/improved. We'll probably even need to pass them when using NostrButton, because probably they won't adapt to all the contexts.